Include/Rocket/Core/StringBase.inl

```diff
@@ -26,42 +26,42 @@
  */
 
 template< typename T >
-StringBase< T >::StringBase() : value((T*)local_buffer), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
+StringBase< T >::StringBase() : value(LocalBuffer()), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
 {
-	value[0] = 0;
+	value[0] = T();
```

Comment:
While this fixes a problem, a simple macro would accomplish the same thing without a function call.  This way assigns a default value, not necessarily null.